File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        n = len(nums)
        sum = 0
        for i in range(0,k):
            sum += nums[i]
        avg = sum/k
        for i in range(k,n):
            sum += nums[i]
            sum -= nums[i-k]
            avg = max(avg,sum/k)
        return avg

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int n = nums.size();
        double sum = 0;
        for(int i=0;i<k;i++)
        {
            sum += nums[i];
        }
        double avg = sum/k;
        for(int i=k;i<n;i++)
        {
            sum += nums[i];
            sum -= nums[i-k];
            avg = max(avg,sum/k);
        }
        return avg;
    }
};